compat/xstring.h

```diff
@@ -10,6 +10,7 @@
 #define SQUID_COMPAT_XSTRING_H
 
 #if HAVE_STRING_H
+#define __STDC_WANT_LIB_EXT1__ 1
```

Comment:
fair enough

Reply:
Depending on what this PR tries to accomplish, we may not want to provide a memset_s() replacement when `__STDC_WANT_LIB_EXT1__` is _not_ defined. The longer you advance this PR without an agreement what it should accomplish, the higher is the risk of redoing or rejecting your work. The scope of this PR may seem obvious to you, but it is not obvious to me, and I cannot guide you without knowing what the end goal is.

Reply:
It s an additional measure to make sure password data is really cleared regardless of the compiler optimisation. It can be pretty sensitive as highlighted here

https://github.com/squid-cache/squid/blob/master/lib/smblib/smbencrypt.c#L171

that is why some oses have implemented their own features (OpenBSD/explicit_bzero, NetBSD/explicit_memset, ...) to address those concerns. I myself had already implemented such in various softwares successfully (php, libb2, ...).

Reply:
Are you implying that the goal of this PR is to make sure (or start making sure) that no-longer-used clear-text password data is wiped clean when it is no longer used?

Reply:
I ve updated both comment and PR commit message.

Reply:
FYI, the PR commit message used by our automation will be the text from your initial submission comment. I have copied the latest commit message to that github comment. Please check that it is correct and edit there if needed.

Reply:
Unless it isn't clear. I am waiting on removal of this #define to approve merge for this PR.